src/services/statistics.service.spec.ts

```diff
@@ -122,4 +133,92 @@ describe('StatisticsService', () => {
       { treatment: 'Corte Regular' },
     );
   });
+
+  describe('getAllTreatments', () => {
+    it('should return an array of treatment names', async () => {
+      const treatmentsMock = [
+        { id: 1, name: 'Corte Regular' },
+        { id: 2, name: 'Tinte' },
+        { id: 3, name: 'Manicure' },
+      ];
+      jest
+        .spyOn(treatmentRepository, 'find')
+        .mockResolvedValue(treatmentsMock as Treatment[]);
+
+      const result = await (service as any).getAllTreatments();
+
+      expect(result).toEqual(['Corte Regular', 'Tinte', 'Manicure']);
+      expect(treatmentRepository.find).toHaveBeenCalledWith({
+        select: ['name'],
+        order: { name: 'ASC' },
+      });
+    });
+  });
+
+  describe('exportStatisticsToExcel', () => {
+    it('should generate an Excel buffer with statistics for all treatments', async () => {
+      const treatmentsMock = [
+        { id: 1, name: 'Corte Regular' },
+        { id: 2, name: 'Tinte' },
+      ];
+
+      jest
+        .spyOn(treatmentRepository, 'find')
+        .mockResolvedValue(treatmentsMock as Treatment[]);
+
+      const getStatisticsMock = jest.spyOn(service, 'getStatistics');
+
+      getStatisticsMock.mockImplementation((start, end, treatment) => {
+        const dto = new ResponseStatisticsDto();
+
+        if (treatment === 'Corte Regular') {
+          dto.totalEarnings = '300';
+          dto.totalServices = '5';
+          dto.totalCompletedAppointments = '4';
+          dto.totalCanceledAppointments = '1';
+        } else if (treatment === 'Tinte') {
+          dto.totalEarnings = '200';
+          dto.totalServices = '3';
+          dto.totalCompletedAppointments = '2';
+          dto.totalCanceledAppointments = '1';
+        } else {
+          dto.totalEarnings = '500';
+          dto.totalServices = '8';
+          dto.totalCompletedAppointments = '6';
+          dto.totalCanceledAppointments = '2';
+        }
+
+        return Promise.resolve(dto);
+      });
+
+      const mockWorksheet = {
+        columns: [],
+        addRow: jest.fn(),
+        getRow: jest.fn().mockReturnValue({
+          getCell: jest.fn().mockReturnValue({ fill: {} }),
+        }),
+      };
+
+      const mockWorkbook = {
+        addWorksheet: jest.fn().mockReturnValue(mockWorksheet),
+        xlsx: {
+          writeBuffer: jest.fn().mockResolvedValue(Buffer.from('test')),
+        },
+      };
+
+      jest
+        .spyOn(ExcelJS, 'Workbook')
+        .mockImplementation(() => mockWorkbook as any);
+
+      const result = await service.exportStatisticsToExcel(
```

Comment:
💭 can you create a spec when any of those two values is missing and that we're successfully validating that scenario please?